src/dnf-plugins/product-id/product-id.c

```diff
@@ -0,0 +1,640 @@
+        return 0;
+    }
+
+    debug("%s v%s, running hook_id: %s on DNF version %d",
+            pinfo.name, pinfo.version, strHookId(id), handle->version);
+
+    if (id == PLUGIN_HOOK_ID_CONTEXT_TRANSACTION) {
+        // Get DNF context
+        DnfContext *dnfContext = handle->initData;
+        // List of all repositories
+        GPtrArray *repos = dnf_context_get_repos(dnfContext);
+        // When there are no repositories, then we can't do anything
+        if(repos == NULL) {
+            return 1;
+        }
+        // List of enabled repositories
+        GPtrArray *enabledRepos = g_ptr_array_sized_new(repos->len);
+        // Enabled repositories with product id certificate
+        GPtrArray *repoAndProductIds = g_ptr_array_sized_new(repos->len);
+        // Enabled repositories with prouctid cert that are actively used
+        GPtrArray *activeRepoAndProductIds = g_ptr_array_sized_new(repos->len);
+
+        ProductDb *productDb = initProductDb();
+        productDb->path = PRODUCTDB_FILE;
+
+        getEnabled(repos, enabledRepos);
+
+        for (guint i = 0; i < enabledRepos->len; i++) {
+            DnfRepo *repo = g_ptr_array_index(enabledRepos, i);
+            LrResult *lrResult = dnf_repo_get_lr_result(repo);
+            LrYumRepoMd *repoMd;
+            GError *tmp_err = NULL;
+
+            debug("Enabled: %s", dnf_repo_get_id(repo));
+            lr_result_getinfo(lrResult, &tmp_err, LRR_YUM_REPOMD, &repoMd);
+            if (tmp_err) {
+                printError("Unable to get information about repository", tmp_err);
+            }
+            else {
+                LrYumRepoMdRecord *repoMdRecord = lr_yum_repomd_get_record(repoMd, "productid");
+                if (repoMdRecord) {
+                    debug("Repository %s has a productid", dnf_repo_get_id(repo));
+                    RepoProductId *repoProductId = (RepoProductId*) malloc(sizeof(RepoProductId));
+                    // TODO: do not fetch productid certificate, when dnf context is set to cache-only mode
+                    // Microdnf not package do not support this feature ATM
+                    gboolean cache_only = dnf_context_get_cache_only(dnfContext);
+                    if (cache_only == TRUE) {
+                        debug("DNF context is set to: cache-only");
+                    } else {
+                        debug("DNF context is NOT set to: cache-only");
+                    }
+                    int fetchSuccess = fetchProductId(repo, repoProductId);
+                    if(fetchSuccess == 1) {
+                        g_ptr_array_add(repoAndProductIds, repoProductId);
+                    } else {
+                        free(repoProductId);
+                    }
+                }
+            }
+        }
+
+        getActive(dnfContext, repoAndProductIds, activeRepoAndProductIds);
+
+        if (activeRepoAndProductIds->len) {
+            // open file
```

Comment:
Is this dead code?

Reply:
It is your WIP code :-). I don't know, what was your intention. I will remove it.